src/features/harmful-apis.js

```diff
@@ -0,0 +1,443 @@
+    if (!permissions || permissions.length === 0) {
+        return
+    }
+    wrapMethod(globalThis.Permissions.prototype, 'query', async function (nativeImpl, queryObject) {
+        // call the original function first in case it throws an error
+        const origResult = await nativeImpl.call(this, queryObject)
+
+        if (permissions.includes(queryObject.name)) {
+            return {
+                name: queryObject.name,
+                state: 'denied',
+                status: 'denied'
+            }
+        }
+        return origResult
+    })
+}
+
+/**
+ * Blocks some privacy harmful APIs.
+ */
+export default class HarmfulApis extends ContentFeature {
+    init (args) {
+        console.log('INIT! from harmfulAPIs', args)
+        // @ts-expect-error linting is not yet seet up for worker context
+        /** @type Navigator | WorkerNavigator */
+        this.navigatorPrototype = globalThis.Navigator?.prototype || globalThis.WorkerNavigator?.prototype
+
+        this.removeDeviceOrientationEvents(this.getFeatureSetting('deviceOrientation'))
+        this.blockGenericSensorApi(this.getFeatureSetting('GenericSensor'))
+        this.filterUAClientHints(this.getFeatureSetting('UaClientHints'))
+        this.removeNetworkInformationApi(this.getFeatureSetting('NetworkInformation'))
+        this.blockGetInstalledRelatedApps(this.getFeatureSetting('getInstalledRelatedApps'))
+        this.removeFileSystemAccessApi(this.getFeatureSetting('FileSystemAccess'))
+        this.blockWindowPlacementApi(this.getFeatureSetting('WindowPlacement'))
+        this.blockWebBluetoothApi(this.getFeatureSetting('WebBluetooth'))
+        this.blockWebUsbApi(this.getFeatureSetting('WebUsb'))
+        this.blockWebSerialApi(this.getFeatureSetting('WebSerial'))
+        this.blockWebHidApi(this.getFeatureSetting('WebHid'))
+        this.blockWebMidiApi(this.getFeatureSetting('WebMidi'))
+        this.removeIdleDetectionApi(this.getFeatureSetting('IdleDetection'))
+        this.removeWebNfcApi(this.getFeatureSetting('WebNfc'))
+        this.filterStorageManagerApi(this.getFeatureSetting('StorageManager'))
+    }
+
+    /**
+     * @param {DeviceOrientationConfig} settings
+     */
+    removeDeviceOrientationEvents (settings) {
+        if (!settings.protected) {
+            return
+        }
+        const eventsToBlock = settings.filterEvents || []
+        if (eventsToBlock.length > 0) {
+            for (const eventName of eventsToBlock) {
+                const dom0HandlerName = `on${eventName}`
+                if (dom0HandlerName in globalThis) {
+                    delete globalThis[dom0HandlerName]
+                }
+            }
+            wrapMethod(EventTarget.prototype, 'addEventListener', function (nativeImpl, type, ...restArgs) {
+                if (eventsToBlock.includes(type) && this === globalThis) {
+                    return
+                }
+                return nativeImpl.call(this, type, ...restArgs)
```

Comment:
Elsewhere we've used `Reflect.apply` this prevents the page overloading `.call` by using the object internals.

Reply:
I've changed it